docs / auth ui (#8511)
supabase/supabase: apps/reference/docs/guides/auth/auth-ui/auth-ui-install.mdx

---
id: auth-ui-install
title: Installation
description: Securing your user data with Row Level Security.
---

# Auth UI

> Auth UI is a React component that provides a simple way to authenticate users.

Reviewer: ```suggestion
Auth UI is a JavaScript library that provides a simple way to authenticate users.
```

Reviewer: You mention React-only below

> It is currently only available for React, however it is planned to be available for other frameworks.

Reviewer: ```suggestion
It currently supports React, however support for other frameworks is planned.
```

> Auth UI relies on [@supabase/supabase-js](/docs/reference/javascript/next/) to work, so you will need to firstly also install [@supabase/supabase-js](/docs/reference/javascript/next/) and perhaps be already familair with that package.

Reviewer: ```suggestion
Auth UI relies on [@supabase/supabase-js](/docs/reference/javascript/next/). You need to also [install @supabase/supabase-js](/docs/reference/javascript/next/installing).
```

### Install via NPM

Install the NPM package from @supabase/react-ui-react

```bash

> npm install @supabase/react-ui-react

Reviewer: ```suggestion
npm install @supabase/react-ui-react @supabase/supabase-js@rc
```

Should we make it easier for them?

Author: oh yea, can do.

```

### Import Auth component

> The most simplest way to use the component is to just pass `supabaseClient` from @supabase/supabase-js as a prop to the component.

Reviewer: ```suggestion
The simplest way to use the component is to pass `supabaseClient` from @supabase/supabase-js as a prop to the component.
```

```js title="/src/index.js"
import { createClient } from '@supabase/supabase-js'
import { Auth } from '@supabase/react-ui-react'

const supabase = createClient(
  '<INSERT PROJECT URL>',
  '<INSERT PROJECT ANON API KEY>'
)

const App = () => <Auth supabaseClient={supabase} />
```

> You may notice that the above renders the Auth component without any styling.

Reviewer: ```suggestion
The example above renders the Auth component without any styling.
```

> It is reccomended that you use one of the provided themes to style the UI.

Reviewer: ```suggestion
We recommend using one of the themes provided to style the UI.
```
